datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyPartitionAssignmentStrategyFactory.java

```diff
@@ -12,69 +12,49 @@
-    VerifiableProperties props = new VerifiableProperties(assignmentStrategyProperties);
-    int cfgMaxTasks = props.getInt(CFG_MAX_TASKS, 0);
-    int cfgImbalanceThreshold = props.getInt(CFG_IMBALANCE_THRESHOLD, 0);
-    int cfgMaxParitionsPerTask = props.getInt(CFG_MAX_PARTITION_PER_TASK, 0);
-    // Set to Optional.empty() if the value is 0
-    Optional<Integer> maxTasks = cfgMaxTasks > 0 ? Optional.of(cfgMaxTasks) : Optional.empty();
-    Optional<Integer> imbalanceThreshold = cfgImbalanceThreshold > 0 ? Optional.of(cfgImbalanceThreshold)
-        : Optional.empty();
-    Optional<Integer> maxPartitions = cfgMaxParitionsPerTask > 0 ? Optional.of(cfgMaxParitionsPerTask) :
-        Optional.empty();
-    boolean enableElasticTaskAssignment = props.getBoolean(CFG_ENABLE_ELASTIC_TASK_ASSIGNMENT,
-        DEFAULT_ENABLE_ELASTIC_TASK_ASSIGNMENT);
-    int cfgPartitionsPerTask = props.getInt(CFG_PARTITIONS_PER_TASK, 0);
-    int cfgPartitionFullnessThresholdPct = props.getIntInRange(CFG_PARTITION_FULLNESS_THRESHOLD_PCT, 0, 0, 100);
-    Optional<Integer> partitionsPerTask = cfgPartitionsPerTask > 0 ? Optional.of(cfgMaxParitionsPerTask) :
-        Optional.empty();
-    Optional<Integer> partitionFullnessThresholdPct = cfgPartitionFullnessThresholdPct > 0 ?
-        Optional.of(cfgPartitionFullnessThresholdPct) : Optional.empty();
-    String cluster = props.getString(CFG_CLUSTER_NAME, null);
+    PartitionAssignmentStrategyConfig config = new PartitionAssignmentStrategyConfig(assignmentStrategyProperties);
 
-    // Create the ZooKeeper Client
+    boolean enableElasticTaskAssignment = config.isElasticTaskAssignmentEnabled();
+    // Create the zookeeper client
     Optional<ZkClient> zkClient = Optional.empty();
-    String zkAddress = props.getString(CFG_ZK_ADDRESS, null);
-    if (enableElasticTaskAssignment && StringUtils.isBlank(zkAddress)) {
-      LOG.warn("Disabling elastic task assignment as zkAddress is not present or empty");
+    try {
+      zkClient = constructZooKeeperClient(enableElasticTaskAssignment, config.getZkAddress(),
+          config.getZkSessionTimeout(), config.getZkSessionTimeout());
+    } catch (IllegalStateException ex) {
+      LOG.warn("Disabling elastic task assignment as zkClient initialization failed");
       enableElasticTaskAssignment = false;
     }
 
+    return new StickyPartitionAssignmentStrategy(config.getMaxTasks(), config.getImbalanceThreshold(),
+        config.getMaxPartitions(), enableElasticTaskAssignment, config.getPartitionsPerTask(),
+        config.getPartitionFullnessThresholdPct(), zkClient, config.getCluster());
+  }
+
+  protected Optional<ZkClient> constructZooKeeperClient(boolean enableElasticTaskAssignment, String zkAddress,
+      int zkSessionTimeout, int zkConnectionTimeout) {
+    Optional<ZkClient> zkClient = Optional.empty();
+    if (enableElasticTaskAssignment && StringUtils.isBlank(zkAddress)) {
```

Comment:
Can you simplify this to 

if (!enableElasticTaskAssignment)
    return Optional.empty() 
if (StringUtils.isBlank(zkAddress)
    throw exception
 return createZkClient(...)